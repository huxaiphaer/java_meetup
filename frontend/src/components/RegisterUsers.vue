<template>
  <v-col class="pa-lg-10">
    <v-card class=" mx-auto">
      <form class="pa-10">
        <p class="reg-text">Java MeetUp Registration</p>
        <v-text-field label="Name" required v-model="name"></v-text-field>
        <v-text-field label="E-mail" required v-model="email"></v-text-field>
        <v-text-field label="Address" required v-model="address"></v-text-field>
        <v-text-field label="Phone Number" v-model="phoneNumber"></v-text-field>
        <v-text-field
          :type="'password'"
          label="Password"
          required
          v-model="password"
        ></v-text-field>
        <v-btn @click="registerUser" class="mr-4" color="primary"
          >Register</v-btn
        >
        <v-btn @click="clear">clear</v-btn>
      </form>
    </v-card>
    <SnackBar />
  </v-col>
</template>

<script>
import { mapActions, mapGetters } from "vuex";
import SnackBar from "./SnackBar";

export default {
  name: "RegisterUsers",
  components: {
    SnackBar
  },
  data() {
    return {
      name: "",
      email: "",
      address: "",
      phoneNumber: "",
      password: "",
      formHasErrors: false
    };
  },
  methods: {
    registerUser() {
      const formData = {
        name: this.name,
        email: this.email,
        address: this.address,
        number: this.phoneNumber,
        password: this.password
      };

      if (!this.validations()) {
        this.register(formData);
      }
    },
    clear() {
      this.name = "";
      this.email = "";
      this.address = "";
      this.phoneNumber = "";
      this.password = "";
    },
    validations() {
      // eslint-disable-next-line no-useless-escape
      const mailFormat = /\S+@\S+\.\S+/;
      const vm = this;
      setTimeout(() => {
        vm.reset_snackbar();
      }, 2000);
      if (this.email === "") {
        this.toast_snackbar_on_error("Email is required");
        return true;
      }

      if (mailFormat.test(this.email.toString()) !== true) {
        this.toast_snackbar_on_error("Please enter a valid mail");
        return true;
      }

      if (this.name === "") {
        this.toast_snackbar_on_error("Name is required");
        return true;
      }
      if (this.address === "") {
        this.toast_snackbar_on_error("Address is required");
        return true;
      }
      if (this.password === "") {
        this.toast_snackbar_on_error("Password  is required");
        return true;
      }
      return this.formHasErrors;
    },
    ...mapActions({
      register: "register/registerUsers",
      reset_snackbar: "register/reset_snackbar",
      toast_snackbar_on_error: "register/toast_snackbar_on_error"
    }),
    computed: {
      ...mapGetters("register", ["snackbar_status"])
    }
  }
};
</script>

<style scoped>
div {
  color: inherit;
}

.reg-text {
  color: black;
  text-align: center;
  font-size: 20px;
  font-weight: bold;
}
</style>
