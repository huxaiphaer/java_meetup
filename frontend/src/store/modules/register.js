import axios from "axios";
import { BASE_URL } from "../../utils/Urls";

export default {
  namespaced: true,
  state: {
    snackbar: {
      snackbar_visibility: false,
      color: "",
      timeout: 0,
      message: ""
    },
    loader: false
  },

  mutations: {
    reset_snackbar(state) {
      state.snackbar.message = "";
      state.snackbar.color = "";
      state.snackbar.timeout = 0;
      state.snackbar.snackbar_visibility = false;
    },

    toast_snackbar_on_success(state) {
      state.snackbar.message = "User registered up Successfully";
      state.snackbar.color = "#4CAF50";
      state.snackbar.timeout = 1000;
      state.snackbar.snackbar_visibility = true;
    },
    toast_snackbar_on_error(state, message) {
      state.snackbar.message = message;
      state.snackbar.color = "#9c4130";
      state.snackbar.timeout = 1000;
      state.snackbar.snackbar_visibility = true;
    },

    loader_on(state) {
      state.loader = true;
    },

    loader_off(state) {
      state.loader = false;
    }
  },
  actions: {
    toast_snackbar_on_error({ commit }, message) {
      commit("toast_snackbar_on_error", message);
    },
    reset_snackbar({ commit }) {
      commit("reset_snackbar");
    },
    registerUsers({ commit }, data) {
      commit("loader_on");
      axios
        .post(`${BASE_URL}/api/v1/users`, data)
        .then(response => {
          commit("loader_on");
          if (response.status == 201) {
            setTimeout(() => {
              commit("reset_snackbar");
            }, 4000);
            commit("toast_snackbar_on_success");
            commit("loader_off");
          } else {
            setTimeout(() => {
              commit("toast_snackbar_on_error", response);
            }, 4000);
            commit("loader_off");
          }
        })
        .catch(error => {
          commit("toast_snackbar_on_error", error.response.data.errors || error.response.data.message)
          commit("loader_off");
        });
    }
  },
  getters: {
    snackbar_status(state) {
      return state.snackbar;
    },
    loader_status(state) {
      return state.loader;
    }
  }
};
