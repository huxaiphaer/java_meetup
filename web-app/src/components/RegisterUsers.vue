<template>

    <v-col class="pa-lg-10">
        <v-card class=" mx-auto">
            <form class="pa-10">
                <p class="reg-text">Java MeetUp Registration</p>
                <v-text-field v-model="name" label="Name" required></v-text-field>
                <v-text-field v-model="email" label="E-mail" required></v-text-field>
                <v-text-field v-model="address" label="Address" required></v-text-field>
                <v-text-field v-model="phoneNumber" label="Phone Number"></v-text-field>
                <v-text-field v-model="password" label="Password" :type="'password'" required></v-text-field>
                <v-btn class="mr-4" color="primary" @click="registerUser">Register</v-btn>
                <v-btn @click="clear">clear</v-btn>
            </form>
        </v-card>
        <SnackBar/>
    </v-col>


</template>

<script>
    import {mapActions, mapGetters} from "vuex";
    import SnackBar from "./SnackBar";

    export default {
        name: "RegisterUsers",
        components: {
            SnackBar
        },
        data() {
            return {
                name: '',
                email: '',
                address: '',
                phoneNumber: '',
                password: '',
                formHasErrors: false,
            }
        },
        methods: {
            registerUser() {
                const formData = {
                    name: this.name,
                    email: this.email,
                    address: this.address,
                    number: this.phoneNumber,
                    password: this.password,
                };

                if (!this.validations()) {
                    this.register(formData);
                }
            },
            clear() {
                this.name = "";
                this.email = "";
                this.address = "";
                this.phoneNumber = "";
                this.password = "";
            },
            validations() {
                // eslint-disable-next-line no-useless-escape
                const mailFormat = /\S+@\S+\.\S+/;
                const vm = this;
                setTimeout(() => {
                    vm.reset_snackbar();
                }, 2000);
                if (this.email === '') {
                    this.toast_snackbar_on_error('Email is required');
                    return true;
                }

                if (mailFormat.test(this.email.toString()) !== true) {
                    this.toast_snackbar_on_error('Please enter a valid mail');
                    return true;
                }

                if (this.name === '') {
                    this.toast_snackbar_on_error('Name is required');
                    return true;
                }
                if (this.address === '') {
                    this.toast_snackbar_on_error('Address is required');
                    return true;
                }
                if (this.password === '') {
                    this.toast_snackbar_on_error('Password  is required');
                    return true;
                }
                return this.formHasErrors;
            },
            ...mapActions({
                register: 'register/registerUsers',
                reset_snackbar: 'register/reset_snackbar',
                toast_snackbar_on_error: 'register/toast_snackbar_on_error',

            }),
            computed: {
                ...mapGetters('register', ['snackbar_status']),
            },

        },
    };
</script>

<style scoped>
    .reg-text {
        color: black;
        text-align: center;
        font-size: 20px;
        font-weight: bold;
    }
</style>
