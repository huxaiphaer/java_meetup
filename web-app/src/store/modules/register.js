import axios from "axios";
import {BASE_URL} from "../../utils/Urls";

export default {
    namespaced: true,
    state: {
        snackbar: {
            snackbar_visibility: false,
            color: "",
            timeout: 0,
            message: ""
        },
        loader: false,
    },

    mutations: {
        reset_snackbar(state) {
            state.snackbar.message = '';
            state.snackbar.color = '';
            state.snackbar.timeout = 0;
            state.snackbar.snackbar_visibility = false;
        },

        toast_snackbar_on_success(state) {
            state.snackbar.message = 'User registered up Successfully';
            state.snackbar.color = '#4CAF50';
            state.snackbar.timeout = 1000;
            state.snackbar.snackbar_visibility = true;
        },
        toast_snackbar_on_error(state, message) {
            state.snackbar.message = message;
            state.snackbar.color = '#9c4130';
            state.snackbar.timeout = 1000;
            state.snackbar.snackbar_visibility = true;
        },

        loader_on(state) {
            state.loader = true;
        },

        loader_off(state) {
            state.loader = false;
        },


    },
    actions: {
        toast_snackbar_on_error({commit}, message) {
            commit('toast_snackbar_on_error', message);
        },
        reset_snackbar({commit}) {
            commit('reset_snackbar');
        },
        registerUsers({commit}, data) {
            commit('loader_on');
            axios
                .post(`${BASE_URL}/api/v1/users`, data)
                .then(response => {
                    if (response.status == 200) {
                        setTimeout(() => {
                            commit('reset_snackbar');
                        }, 2000);
                        commit('toast_snackbar_on_success');
                        commit('loader_off');

                    } else if (response.status == 400) {
                        console.log(" whats here  " + response)
                        commit('toast_snackbar_on_error', response);
                        commit('loader_off');
                    }
                })
                .catch(error => {
                    console.log(" error " + error.response.data)
                    error.response.data.errors.map((t) => {
                        commit('toast_snackbar_on_error', t);
                    })
                    commit('loader_off');
                });
        }
    },
    getters: {
        snackbar_status(state) {
            return state.snackbar;
        },
        loader_status(state) {
            return state.loader;
        },

    }
};
