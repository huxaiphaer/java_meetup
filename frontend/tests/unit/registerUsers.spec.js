import RegisterUsers from "@/components/RegisterUsers";
import register from '@/store/modules/register'
import {createLocalVue, mount, shallowMount} from "@vue/test-utils"
import Vuex from "vuex"
import Vuetify from 'vuetify'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)


describe("Register.vue", () => {

    let vuetify
    let store;
    beforeEach(() => {
        store = new Vuex.Store({modules: {register}})
    })

    it("should have a regsiterUsers component and match snapshot", () => {

        const wrapper = shallowMount(RegisterUsers, {
            vuetify,
            store,
            localVue
        })
        expect(wrapper.html()).toMatchSnapshot()
    });

    it("should expect an error when missing Name", async () => {

        const wrapper = mount(RegisterUsers, {
            localVue,
            vuetify,
            store,

        });

        wrapper.setData({
            name: '',
            email: 'ab@gmail.com',
            address: 'Buziga, Kampala',
            phoneNumber: '0704594180',
            password: '9393939',
            formHasErrors: false,
        });

        const button = wrapper.find({name: 'v-btn'})
        button.trigger('click');
        await wrapper.vm.$nextTick()
    });

    it("should expect an error when missing email", async () => {
        const wrapper = mount(RegisterUsers, {
            localVue,
            vuetify,
            store,

        });

        wrapper.setData({
            name: 'huxy',
            email: '',
            address: 'Buziga, Kampala',
            phoneNumber: '0704594180',
            password: '9393939',
            formHasErrors: false,
        });

        const button = wrapper.find({name: 'v-btn'})
        button.trigger('click');
        await wrapper.vm.$nextTick()
    })

    it("should expect an error when missing address", async () => {
        const wrapper = mount(RegisterUsers, {
            localVue,
            vuetify,
            store,

        });

        wrapper.setData({
            name: 'huxy',
            email: 'huxy@gmail.com',
            address: '',
            phoneNumber: '0704594180',
            password: '9393939',
            formHasErrors: false,
        });

        const button = wrapper.find({name: 'v-btn'})
        button.trigger('click');
        await wrapper.vm.$nextTick()
    })

    it("should expect an error when missing phone number", async () => {
        const wrapper = mount(RegisterUsers, {
            localVue,
            vuetify,
            store,

        });

        wrapper.setData({
            name: 'huxy',
            email: 'huxy@gmail.com',
            address: 'Buziga',
            phoneNumber: '',
            password: '9393939',
            formHasErrors: false,
        });

        const button = wrapper.find({name: 'v-btn'})
        button.trigger('click');
        await wrapper.vm.$nextTick()
    })

    it("should expect an error when missing password", async () => {
        const wrapper = mount(RegisterUsers, {
            localVue,
            vuetify,
            store,
        });

        wrapper.setData({
            name: 'huxy',
            email: 'huxy@gmail.com',
            address: 'Buziga',
            phoneNumber: '0896876786',
            password: '',
            formHasErrors: false,
        });

        const button = wrapper.find({name: 'v-btn'})
        button.trigger('click');
        await wrapper.vm.$nextTick()
    })
    it("should expect an error when missing email doesn't meet conditions", async () => {
        const wrapper = mount(RegisterUsers, {
            localVue,
            vuetify,
            store,

        });

        wrapper.setData({
            name: 'huxy',
            email: 'huxygmail',
            address: 'Buziga',
            phoneNumber: '0896876786',
            password: '737738hdhygfuy',
            formHasErrors: false,
        });

        const button = wrapper.find({name: 'v-btn'})
        button.trigger('click');
        await wrapper.vm.$nextTick()
    })
});



