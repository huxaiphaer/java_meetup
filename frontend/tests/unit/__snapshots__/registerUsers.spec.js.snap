// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue should have a regsiterUsers component and match snapshot 1`] = `
<v-col-stub tag="div" class="pa-lg-10">
  <v-card-stub loaderheight="4" tag="div" class=" mx-auto">
    <form class="pa-10">
      <p class="reg-text">Java MeetUp Registration</p>
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="Name" loaderheight="2" clearicon="$clear" type="text" required=""></v-text-field-stub>
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="E-mail" loaderheight="2" clearicon="$clear" type="text" required=""></v-text-field-stub>
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="Address" loaderheight="2" clearicon="$clear" type="text" required=""></v-text-field-stub>
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="Phone Number" loaderheight="2" clearicon="$clear" type="text"></v-text-field-stub>
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="Password" loaderheight="2" clearicon="$clear" type="password" required=""></v-text-field-stub>
      <v-btn-stub color="primary" tag="button" activeclass="" type="button" class="mr-4">Register</v-btn-stub>
      <v-btn-stub tag="button" activeclass="" type="button">clear</v-btn-stub>
    </form>
  </v-card-stub>
  <snackbar-stub></snackbar-stub>
</v-col-stub>
`;
